media/CircleEditor/view-sidebar.js

```diff
@@ -1456,30 +1439,269 @@ sidebar.ArgumentView = class {
         return true;
     }
 
-    /**
-     * TODO Implement compareData()
-     * Compare and change transformed data against corrected data
-     */
+    compareData(bufferArr, originalArr, originalBuf, arrType) {
+        let dataLength = arrType === 4 ? bufferArr.length * 2 : bufferArr.length;
+        for (let i = 0; i < dataLength; i++) {
+            /* compare string formatted buffer data */
+            if (originalArr[i] !== bufferArr[i]) {
+                /* when data change detected, update buffer according to type */
+                var buffer = new ArrayBuffer(8);
+                const view = new DataView(buffer);
+                switch (arrType) {
+                    case 'float32':
+                        view.setFloat32(0, parseFloat(bufferArr[i]), true);
+                        for (let j = 0; j < 4; j++) {
+                            originalBuf[i * 4 + j] = view.getUint8(j);
+                        }
+                        break;
+                    case 'float16':
+                        view.setFloat16(0, parseFloat(bufferArr[i]), true);
+                        for (let j = 0; j < 2; j++) {
+                            originalBuf[i * 2 + j] = view.getUint8(j);
+                        }
+                        break;
+                    case 'int32':
+                        view.setInt32(0, parseInt(bufferArr[i]), true);
+                        for (let j = 0; j < 4; j++) {
+                            originalBuf[i * 4 + j] = view.getUint8(j);
+                        }
+                        break;
+                    case 'uint8':
+                        view.setUint8(0, parseUint(bufferArr[i]), true);
+                        originalBuf[i] = view.getUint8(0);
+                        break;
+                    case 'int64':
+                        view.setBigInt64(0, BigInt(parseInt(bufferArr[i])), true);
+                        for (let j = 0; j < 4; j++) {
+                            originalBuf[i * 4 + j] = view.getUint8(j);
+                        }
+                        break;
+                    case 'boolean':
+                        if (bufferArr[i] === 'false' || bufferArr[i] - 0 === 0) {
+                            originalBuf[i] = 0;
+                        } else {
+                            originalBuf[i] = 1;
+                        }
+                        break;
+                    case 'int16':
```

Comment:
Is there any reason to omit `uint16`? :)

Reply:
Please refer to the comment above!